README.md

```diff
@@ -2,54 +2,113 @@
 
 # Terraform provider for Ironic
 
-This is a terraform provider that lets you provision baremetal servers managed by Ironic.
+## Provider declaration
 
-# Usage
-
-Example:
+Currently the provider only supports stand-alone noauth Ironic.  At a minimum, the Ironic
+endpoint URL must be specified. The user may also optionally specify an API microversion.
 
 ```terraform
 provider "ironic" {
-  "url" = "http://localhost:6385/v1"
-  "microversion" = "1.50"
+  url          = "http://localhost:6385/v1"
+  microversion = "1.50"
 }
+```
+
+## Resources
+
+This provider currently implements 3 resources: `ironic_node_v1`, `ironic_port_v1`, and
+`ironic_deployment`.  The first two correspond directly to objects in Ironic's API, 
+but a deployment is a concept specific to the provider. It operates similar to a
+[Metalsmith deployment](https://docs.openstack.org/metalsmith/latest/), where a
+deployment consumes a node.  This allows a user to manage their hardware outside of
+terraform -- perhaps through Ironic's discovery mechanism.
+
+## Nodes
 
+A node describes a hardware resource. If a node of the same name already exists, or
+`node_uuid` is specified, it's details would be updated. This could be used to provide
+BMC information for a discovered node.
+
+Destruction of a node object removes the node from Ironic.
+
+
+```
 resource "ironic_node_v1" "openshift-master-0" {
   name = "openshift-master-0"
-  target_provision_state = "active"
-  user_data = "${file("master.ign")}"
 
   ports = [
     {
-      "address" = "00:bb:4a:d0:5e:38"
-      "pxe_enabled" = "true"
-    }
+      pxe_enabled = true
+      address     = "00:bb:4a:d0:5e:38"
+    },
   ]
-
-  properties {
-    "local_gb" = "50"
```

Comment:
`local_gb` and `cpu_arch` should stay in this resource for sure (maybe not nested in properties). I think we should optionally support `root_disk` here as well.